util/topsql/reporter/reporter.go

```diff
@@ -341,6 +350,34 @@ func (tsr *RemoteTopSQLReporter) collectWorker() {
 	}
 }
 
+func (tsr *RemoteTopSQLReporter) handleClientRegistry() {
+	tsr.clientRegistry.visitAndClear(func(client ReportClient) {
+		tsr.clients = append(tsr.clients, client)
+	})
+
+	idx := 0
+	pendingCnt := 0
+	for _, client := range tsr.clients {
```

Comment:
Deserve a comment documentating that it will remove all clients that is down state.